src/manage_nsfs/health.js

```diff
@@ -421,6 +425,38 @@ class NSFSHealth {
             err_obj
         };
     }
+
+    /**
+     * _get_config_dir_status returns the config directory phase, version and equivalent package_version
+     * @param {Object} system_data 
+     * @returns {Object}
+     */
+    _get_config_dir_status(system_data) {
+        if (!system_data) return { error: 'system data is missing' };
+        const config_dir_data = system_data.config_directory;
+        if (!config_dir_data) return { error: 'config directory data is missing' };
+        return {
+            phase: config_dir_data.phase,
+            config_dir_version: config_dir_data.config_dir_version,
+            upgrade_package_version: config_dir_data.upgrade_package_version,
+            in_progress_upgrade: config_dir_data.in_progress_upgrade
+        };
+    }
+
+    /**
+     * get_upgrade_status returns the status of an ongoing upgrade, if valid it returns an object with upgrade details
+     * if upgrade is not ongoing but config dir is locked in one of the hosts, the error details will return
```

Comment:
Where do we return the "an object with upgrade details" I don't see it?

Reply:
hmm maybe it was missing when you reviewed it, it's there now